src/thor/trace_route_action.cc

```diff
@@ -145,12 +145,13 @@ std::vector<std::tuple<float, std::vector<thor::MatchResult>, odin::TripPath>> t
   // Call Meili for map matching to get a collection of pathLocation Edges
   matcher->set_interrupt(interrupt);
   // Create the vector of matched path results
-  std::vector<std::vector<meili::MatchResult>> offline_results;
+  std::map<float, std::vector<meili::MatchResult>> offline_results;
   if (trace.size() > 0)
     offline_results = matcher->OfflineMatch(trace, best_paths);
 
   // Process each score/match result
-  for (const auto& match_results : offline_results) {
+  for (const auto& scored_match_results : offline_results) {
+    const auto& match_results = scored_match_results.second;
     std::vector<thor::MatchResult> enhanced_match_results;
     odin::TripPath trip_path;
     std::unordered_map<size_t, std::pair<RouteDiscontinuity, RouteDiscontinuity>> route_discontinuities;
```

Comment:
@ptpt could we do the penalty down in here where we determine where the discontinuities are. i was thinking that i could based off of the mode and distance between the two points on either side of the discontinuity i could compute a penality? does that make sense at all? what is your preference?